ui/src/app/layout/dashboard/dashboard.component.ts

```diff
@@ -108,13 +108,38 @@ export class DashboardComponent implements OnInit {
                 change.change.html = this.changeHtml(change.change);
                 return change;
             }).filter(change => change.change.html !== null);
+
         const index = changes.findIndex(
             change =>
                 new Date(change.timestamp) <
                 new Date(new Date().setDate(new Date().getDate() - minDays)));
-        return changes.slice(0, Math.max(amountOfEntries, index));
+
+        // eslint-disable-next-line @typescript-eslint/prefer-for-of
+        let keeper = [changes[0]];
+        for (let i = 0; i < changes.length; i++) {
+            keeper.push(this.detectMultipleJsonSchemaChangesInARow(changes,i,i+1));
+        }
+        keeper = keeper.filter(keep => keep !== null);
+        return keeper.slice(0, Math.max(amountOfEntries, index));
     }
 
+    private detectMultipleJsonSchemaChangesInARow(changes: ChangelogEntry[], indexOne: number, indexTwo: number) {
+        if (
```

Comment:
This code looks as if it would fail in the following scenario:
- Only the previous DELETION of a schema is visible in the changes (because the associated creation is too old)
- Now, I add a new schema
- Result: Both is removed from changelog, I do not see the new, currently valid, schema in the changelog

This is only from reading the code, I may be wrong here.

Reply:
I think you are wrong. The function is called on every two pairs of entries (0 and 1, 1 and 2, 2 and 3). This means that the function will be called as many times as there are entries in the changelog. Let's take the example that we have this changelog:
[0] create topic - June 23
[1] Delete schema - June 21
[2] Create schema - June 19
[X] .....
[100] Delete schema - Jan 3

For example, if only the first 20 entries are displayed in the changelog, entry 100 will not be displayed. Entry zero is always shown in the changelog (line 118: let keeper = [changes[0]];). Now the function is called for the first time with element zero and element one. Since these two are not schemas, the element one is now also added to the changelog. The next call will check item one and item two. These are both about schemas and of the same subject => the function returns null. 

This leads to the desired result: only the newest entry for schemas is displayed if they are created and deleted one after the other without any other event in between. 

This deletes the entry without regard to how far apart they were.